src/ocean/task/Task.d

```diff
@@ -456,8 +456,39 @@ public abstract class Task : ISuspendable
         debug_trace("<{}> start of main function", cast(void*) this);
 
         scope(exit)
+        {
             this.state_bitmask |= TaskState.Finished;
 
+            if (this.termination_hooks.length)
```

Comment:
This seems like a potentially dangerous change: running the termination hooks inside the task, rather than from the calling context.

Reply:
There is no change, it already works like that (the fiber is the same).

Reply:
Ah I see, `runTask` is called from inside the fiber. :+1: